tests/dofs/dof_accessor_04.cc

```diff
@@ -0,0 +1,86 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2021 - 2023 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+// Test DoFCellAccessor::distribute_local_to_global(
+// local_matrix, local_vector, global_matrix, global_vector) 
+
+#include <deal.II/grid/tria.h>
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/dofs/dof_tools.h>
+#include <deal.II/fe/fe_q.h>
+#include <deal.II/lac/dynamic_sparsity_pattern.h>
+#include <deal.II/lac/sparse_matrix.h>
+#include <deal.II/lac/full_matrix.h>
+#include <deal.II/lac/vector.h>
+
+#include <iostream>
+
+int main()
+{
+  using namespace dealii;
+
+  // create the triangulation
+  Triangulation<2> tria;
+  GridGenerator::hyper_cube(tria, -1, 1);
+  tria.refine_global(1);
+
+  // distribute dofs
+  DoFHandler<2> dof_handler(tria);
+
+  FE_Q<2> fe(1);
+  dof_handler.distribute_dofs(fe);
+
+  // create the global matrix and global vector
+  DynamicSparsityPattern dsp(dof_handler.n_dofs());
+  DoFTools::make_sparsity_pattern(dof_handler, dsp);
+  SparsityPattern sp;
+  sp.copy_from(dsp);
+
+  SparseMatrix<double> global_matrix(sp);
+  Vector<double> global_vector(dof_handler.n_dofs());
+
+  // create local matrix and local vector
+  FullMatrix<double> local_matrix(fe.dofs_per_cell, fe.dofs_per_cell);
+  Vector<double> local_vector(fe.dofs_per_cell);
+  for (unsigned int i = 0; i < fe.dofs_per_cell; ++i)
+  {
+    local_vector(i) = i + 1;
+    for (unsigned int j = 0; j < fe.dofs_per_cell; ++j)
+      local_matrix(i, j) = i * fe.dofs_per_cell + j + 1;
+  }
+
+  // call function distribute_local_to_global()
+  auto cell = dof_handler.begin_active();
+  cell->distribute_local_to_global(local_matrix, local_vector, global_matrix, global_vector);
+
+  // output the local matrix and local vector
+  std::cout << "local matrix:" << std::endl;
+  local_matrix.print_formatted(std::cout, 0, false);
+
+  std::cout << "local vector:" << std::endl;
+  local_vector.print(std::cout, 0, false);
+  
+  std::cout << std::endl;
+
+  // output the global matrix and global vector
+  std::cout << "global matrix:" << std::endl;
+  global_matrix.print_formatted(std::cout, 0, false);
+
+  std::cout << "global vector:" << std::endl;
+  global_vector.print(std::cout, 0, false);
```

Comment:
```suggestion
  deallog << "local matrix:" << std::endl;
  local_matrix.print_formatted(deallog.get_file_stream(), 0, false);

  deallog << "local vector:" << std::endl;
  local_vector.print(deallog.get_file_stream(), 0, false);
  
  deallog << std::endl;

  // output the global matrix and global vector
  deallog << "global matrix:" << std::endl;
  global_matrix.print_formatted(deallog.get_file_stream(), 0, false);

  deallog << "global vector:" << std::endl;
  global_vector.print(deallog.get_file_stream(), 0, false);
```

Reply:
Thanks for the suggestions. I have revised the indentation by "clang-format". However, functions SparseMatrix::print_formatted(), FullMatrix::print_formatted() and Vector::print() cannot recognize dealii::Logstream. To avoid calling these functions, I modified the outputs by checking if each entry of the local matrix/vector has found its position in the global matrix/vector.

Reply:
Have you tried using `deallog.get_file_stream()` as suggested? In other parts of the testsuite we do exactly that, see for example: https://github.com/dealii/dealii/blob/22ec4132bcbc0d6b939cb94f9170c8720bc176dd/tests/lapack/full_matrix_30.cc#L39

I am fine with either way you designed the test :)

Reply:
Sorry for reading your codes carelessly... I have changed the test back to its original version, which I do prefer :-) The updated test has been checked and properly formatted.